sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -212,3 +220,123 @@ Java_io_sentry_android_ndk_SentryNdk_initSentryNative(JNIEnv *env, jclass cls,
 
     sentry_init(options);
 }
+
+JNIEXPORT void JNICALL
+Java_io_sentry_android_ndk_NativeModuleListLoader_nativeClearModuleList(JNIEnv *env, jclass cls) {
+    sentry_clear_modulecache();
+}
+
+JNIEXPORT jobjectArray JNICALL
+Java_io_sentry_android_ndk_NativeModuleListLoader_nativeLoadModuleList(JNIEnv *env, jclass cls) {
+    sentry_value_t image_list_t = sentry_get_modules_list();
+    jobjectArray image_list = NULL;
+
+    if (sentry_value_get_type(image_list_t) == SENTRY_VALUE_TYPE_LIST) {
+        size_t len_t = sentry_value_get_length(image_list_t);
+
+        jclass image_class = (*env)->FindClass(env, "io/sentry/protocol/DebugImage");
+        image_list = (*env)->NewObjectArray(env, len_t, image_class, NULL);
+
+        jmethodID image_addr_method = (*env)->GetMethodID(env, image_class, "setImageAddr",
+                                                          "(Ljava/lang/String;)V");
+
+        jmethodID image_size_method = (*env)->GetMethodID(env, image_class, "setImageSize",
+                                                          "(J)V");
+
+        jmethodID code_file_method = (*env)->GetMethodID(env, image_class, "setCodeFile",
+                                                         "(Ljava/lang/String;)V");
+
+        jmethodID image_addr_ctor = (*env)->GetMethodID(env, image_class, "<init>",
+                                                        "()V");
+
+        jmethodID type_method = (*env)->GetMethodID(env, image_class, "setType",
+                                                    "(Ljava/lang/String;)V");
+
+        jmethodID debug_id_method = (*env)->GetMethodID(env, image_class, "setDebugId",
+                                                        "(Ljava/lang/String;)V");
+
+        jmethodID code_id_method = (*env)->GetMethodID(env, image_class, "setCodeId",
+                                                       "(Ljava/lang/String;)V");
+
+        jmethodID debug_file_method = (*env)->GetMethodID(env, image_class, "setDebugFile",
+                                                          "(Ljava/lang/String;)V");
+
+        for (size_t i = 0; i < len_t; i++) {
+            sentry_value_t image_t = sentry_value_get_by_index(image_list_t, i);
+
+            if (!sentry_value_is_null(image_t)) {
+                jobject image = (*env)->NewObject(env, image_class, image_addr_ctor);
```

Comment:
when/where is this freed?

Reply:
after talking to @Swatinem this was not necessary, maybe after #1050 this is worth another review?
@Swatinem mind reviewing this part again?

Reply:
It looks fine from my side. TBH, I have no idea how the ownership transfer works for these JNI object, and if the GC will ever run concurrently with this call? If we would need to manually root these things on the C stack, or if we need to *unref* them rather.